source/Engine.cpp

```diff
@@ -1331,6 +1348,9 @@ void Engine::CalculateStep()
 	// missile systems ready to fire.
 	hasAntiMissile.clear();
 	
+	// See if any new system weather has started and perform any active weather events.
+	DoWeather();
+	
```

Comment:
Is it intended that this occurs after the other "generate new things that happen this step" actions?